lib/active_record/connection_adapters/odbc_adapter.rb

```diff
@@ -167,10 +167,13 @@ def initialize_type_map(m = type_map)
         m.alias_type 'bool', 'boolean'
         m.alias_type 'varbinary', 'binary'
         m.alias_type 'variant', 'json'
-        m.alias_type 'object', 'json'
-        m.alias_type 'array', 'json'
+        m.alias_type 'object', 'string'
+        m.alias_type 'array', 'string'
         m.alias_type 'geography', 'char'
         m.alias_type 'geometry', 'char'
+
+        # number() data types in Snowflake are interpreted as decimal and must be mapped back to a float
+        m.alias_type 'decimal',    'float' if @database_metadata&.dbms_name&.downcase == "snowflake"
```

Comment:
I think just override by adding comment, condition not needed

Reply:
Removed the condition.